sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -0,0 +1,1095 @@
+
+                eval("self._widget.lcdE%02d_%d.display(%d)" % (n + 1, self.biotac_name, value))
+                colour = self.get_electrode_colour_from_value(value)
+
+                elipse_x = self.sensing_electrodes_x[n]
+                elipse_y = self.sensing_electrodes_y[n]
+
+                if n < 9:
+                    text_x = elipse_x + self.x_display_offset[1]
+                    text_y = elipse_y + self.y_display_offset[1]
+
+                else:
+                    text_x = elipse_x + self.x_display_offset[2]
+                    text_y = elipse_y + self.y_display_offset[2]
+
+                self.draw_electrode(painter, elipse_x, elipse_y, text_x, text_y,
+                                    colour, str(n + 1))
+
+        painter.setFont(QFont("Arial", self.label_font_size[1]))
+
+        for n in range(len(self.excitation_electrodes_x)):
+            elipse_x = self.excitation_electrodes_x[n]
+            elipse_y = self.excitation_electrodes_y[n]
+
+            colour = QColor(127, 127, 127)
+
+            text_x = elipse_x + self.x_display_offset[3]
+            text_y = elipse_y + self.y_display_offset[3]
+
+            self.draw_electrode(painter, elipse_x, elipse_y, text_x, text_y,
+                                colour, "X" + str(n + 1))
+
+        self._widget.update()
+
+    def subscribe_to_topic(self, prefix):
+        if prefix:
+            rospy.Subscriber(prefix + "tactile", BiotacAll, self.tactile_cb)
+
+    def load_params(self):
+
+        self.RECTANGLE_WIDTH = rospy.get_param(
+            "sr_gui_biotac/electrode_display_width",
+            30)  # Display sizes for electrodes in pixels
+        self.RECTANGLE_HEIGHT = rospy.get_param(
+            "sr_gui_biotac/electrode_display_height", 30)
+
+        self.factor = 5
+        # location on the sensor in mm to display location in pixels
+        self.x_display_offset = rospy.get_param(
+            "sr_gui_biotac/x_display_offset", [125, 12.5, 4.5,
+                                               3.5])  # Pixel offsets for
+        # displaying electrodes. offset[0] is applied to each electrode.
+        # 1,2 and 3 are the label offsets for displaying electrode number.
+        self.y_display_offset = rospy.get_param(
+            "sr_gui_biotac/y_display_offset", [-30, 4.0, 4.0, 4.0])
+        self.label_font_size = rospy.get_param(
+            "sr_gui_biotac/electrode_label_font_sizes", [8,
+                                                         6])  # Font sizes
+        # for labels on sensing + excitation electrodes
+
+        if self._hand_parameters.mapping:
+            self.default_topic = (
+                    self._hand_parameters.mapping.values()[0] + '/')
+        else:
+            self.default_topic = ""
+
+    def find_children(self, biotac_name):
+        self.widget = self._widget.findChild(QWidget, "widget" + "_" + str(biotac_name))
+        self.biotac_name = biotac_name
+
+    def __init__(self, context):
```

Comment:
It would be nicer to move this to the beginning of the class

Reply:
Done